src/docfx/build/page/BuildPage.cs

```diff
@@ -187,6 +187,11 @@ private static (List<Error>, SystemMetadata) CreateSystemMetadata(Context contex
             systemMetadata.SearchProduct = context.Config.Product;
             systemMetadata.SearchDocsetName = context.Config.Name;
 
+            if (IsCustomized404Page(file))
+            {
+                systemMetadata.Robots = "NOINDEX, NOFOLLOW";
```

Comment:
Do we have to add the metadata pragmatically? The true page https://github.com/MicrosoftDocs/DocsRoot/blob/a5d336d517b20b65af10d57ce831818767b22adb/DocsCoreContent/404.md already has configured `noindex, nofollow` as metadata.

Reply:
At least, it is feature of v2, and there are some 404 files([bagie/docs](https://bagie.visualstudio.com/DefaultCollection/tools/_git/docs?path=%2FCTAS%2F404.md),  [MicrosoftDocs/msrd-dev-center-pr](https://github.com/MicrosoftDocs/msrd-dev-center-pr/blob/master/articles/404.md)) which doesn't have this metadata

Reply:
I see. How about moving line 51 down here so the check, report, action happens at the same place?

Reply:
Yeah... I have thought about that, but do you think that is a little bit wired to report this warning in a function called `CreateSystemMetadata`. I can do the move if you think that would be better~

Reply:
Yes, it is better, `CreateSystemMetadata` already reports tons of warnings.